File: AkkaCustomExtension/src/main/java/org/akka/essentials/custom/extension/example/TestApp.java
package org.akka.essentials.custom.extension.example;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;

import com.typesafe.config.ConfigFactory;

public class TestApp {

	public TestApp() {

		ActorSystem _system = ActorSystem.create("Extension-Test",
				ConfigFactory.load().getConfig("TestApp"));

		ActorRef ref = _system.actorOf(new Props(MyActor.class));
		
		ref.tell("msg");

		MySQLJDBCSettingsImpl mysqlSetting = MySQLJDBCSettings.SettingsProvider
				.get(_system);

		System.out.println(mysqlSetting.DB_NAME);
		System.out.println(mysqlSetting.DB_URL);

		_system.shutdown();
	}

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		new TestApp();

	}

}

File: AkkaCustomExtension/src/main/scala/org/akka/essentials/custom/extension/example/TestApp.scala
package org.akka.essentials.custom.extension.example

import com.typesafe.config.ConfigFactory

import akka.actor.actorRef2Scala
import akka.actor.ActorSystem
import akka.actor.Props

object TestApp {

  def main(args: Array[String]): Unit = {
    val system = ActorSystem("Extension-Test", ConfigFactory.load()
      .getConfig("TestApp"))

    val ref = system.actorOf(Props[MyActor])
    ref ! "print"

    val mysqlSetting = MySQLJDBCSettings(system)

    println(mysqlSetting.DB_NAME)
    println(mysqlSetting.DB_URL)
  }

}